site/tests/integration/blockpage.test.ts

```diff
@@ -0,0 +1,113 @@
+);
+
+if (!codeBlock || !unsupportedBlock) {
+  throw new Error("Code and Embed blocks need to be prepared before tests run");
+}
+
+test("Block page should contain key elements", async ({ page, isMobile }) => {
+  await page.goto(codeBlock.blockPackagePath);
+
+  await expect(
+    page.locator(`h1:has-text('${codeBlock.displayName}')`),
+  ).toBeVisible();
+
+  await expect(
+    page.locator(`p:has-text('${codeBlock.description}')`),
+  ).toBeVisible();
+
+  await expect(page.locator(`text=@${codeBlock.author}`)).toBeVisible();
+  await expect(page.locator(`text=@${codeBlock.author}`)).toHaveAttribute(
+    "href",
+    `/@${codeBlock.author}`,
+  );
+  await expect(page.locator(`text=V${codeBlock.version}`)).toBeVisible();
+
+  const blockExample = codeBlock.examples[0] as Record<string, string>;
+
+  if (isMobile) {
+    await page.locator("text=Source Code").click();
+    await expect(page.locator("text=Data Source")).toBeVisible();
+  } else {
+    await expect(page.locator("text=Block Properties")).toBeVisible();
+  }
+  await expect(page.locator("#simple-tabpanel-0 >> textarea")).toHaveValue(
+    JSON.stringify(blockExample, null, 2),
+  );
+
+  if (isMobile) {
+    await page.locator("text=Data Source").click();
+    await page.locator("text=Block Schema").click();
+  } else {
+    await page.locator("text=Block Schema").click();
+  }
+  await expect(page.locator("#simple-tabpanel-1")).toBeVisible();
+  // await expect(page.locator("#simple-tabpanel-1")).not.toBeEmpty();
+
+  if (isMobile) {
+    await page.locator("text=Preview").click();
+  }
+
+  await expect(page.locator("iframe[title='block']")).toBeVisible();
+  await expect(
+    page.frameLocator("iframe[title='block']").locator("input"),
+  ).toHaveValue(blockExample.caption!);
+
+  // check if readme was displayed
+  await expect(page.locator("article")).toBeVisible();
+
+  await expect(page.locator("h2:has-text('Repository')")).toBeVisible();
+
+  await expect(
+    page.locator("a:below(h2:has-text('Repository'))").first(),
+  ).toHaveAttribute("href", codeBlock.repository);
+
+  await expect(page.locator("text=Explore more blocks")).toBeVisible();
+
+  // Block slider
+  await expect(page.locator('[data-testid="block-slider"]')).toBeVisible();
+
+  // New blocks can get added, hence the usage of greater than instead of equal too
+  expect(
+    await page.locator('[data-testid="block-slider"] >> .slick-slide').count(),
+  ).toBeGreaterThan(5);
+});
+
+test("should show an error message if an unsupported block is rendered", async ({
+  page,
+  isMobile,
+}) => {
+  await page.goto(unsupportedBlock.blockPackagePath);
+
+  const blockExample = unsupportedBlock.examples[0] as Record<string, string>;
+
+  if (isMobile) {
+    await page.locator(".MuiTabs-root >> text=Source Code").click();
+  }
+
+  // confirm block properties tab contains example data
+  await expect(page.locator("#simple-tabpanel-0 >> textarea")).toHaveValue(
```

Comment:
Optional: replace classes and ids with data-testid. This will make selectors look more semantically meaningful and less fragile when we change implementation details. For example, if we update MUI components and their class names have changed, we should not expect the tests to fail as long as the app looks the same for the end users.